src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs

```diff
@@ -205,7 +205,17 @@ public static Assembly LoadWithPartialName(string partialName)
             if (partialName == null)
                 throw new ArgumentNullException(nameof(partialName));
 
-            return Load(partialName);
+            if ((partialName.Length == 0) || (partialName[0] == '\0'))
```

Comment:
`string.IsNullOrEmpty`

Reply:
Also - I think the Length will be zero if the first character is the nul terminator.